front-end/app/checkout/checkout.component.ts

```diff
@@ -106,4 +106,11 @@ export class CheckoutComponent {
     isOrderButtonDisplayed(): boolean {
         return this.deliveryMethod !== null && this.paymentMethod !== null;
     }
+
```

Comment:
I don't think ngDoCheck is the best way of doing this. It would probably be better to change:
`billingAddress: this.billingAddressSameAsDeliveryAddress ? this.deliveryAddress : this.billingAddress,`
To this:
`billingAddress: this.isAddressDisplayed() && this.billingAddressSameAsDeliveryAddress ? this.deliveryAddress : this.billingAddress,`